x-pack/plugins/apm/public/components/app/ServiceMap/Popover/ServiceMetricFetcher.tsx

```diff
@@ -35,7 +49,52 @@ export function ServiceMetricFetcher({
       preservePreviousData: false,
     }
   );
-  const isLoading = status === 'loading';
 
-  return <ServiceMetricList {...data} isLoading={isLoading} />;
+  const isLoading =
+    status === FETCH_STATUS.PENDING || status === FETCH_STATUS.LOADING;
+
+  if (isLoading) {
+    return <LoadingSpinner />;
+  }
+
+  if (environment && !data.hasEnvironmentData) {
+    return (
+      <EuiCallOut
+        title={i18n.translate(
+          'xpack.apm.serviceMap.popoverMetrics.noEnvironmentDataCallout.title',
+          {
+            defaultMessage: 'No data for current environment',
+          }
+        )}
+        size="s"
+        iconType="iInCircle"
+      >
+        {i18n.translate(
+          'xpack.apm.serviceMap.popoverMetrics.noEnvironmentDataCallout.text',
+          {
+            defaultMessage: `No data available. Try switching to another environment.`,
```

Comment:
I would remove "No data available" from this text as it's just a repeat of the title

```suggestion
            defaultMessage: `Try switching to another environment.`,
```